processor/processor-configuration/src/main/java/com/redhat/service/smartevents/processor/ProcessorCatalogServiceImpl.java

```diff
@@ -116,7 +116,7 @@ public List<String> getSourcePasswordProperties(String id) {
     }
 
     private String readFile(String resourceDirectory, String name) {
-        InputStream is = Thread.currentThread().getContextClassLoader().getResourceAsStream(resourceDirectory + name + JSON_FILE_EXTENSION);
+        InputStream is = this.getClass().getResourceAsStream(resourceDirectory + name + JSON_FILE_EXTENSION);
```

Comment:
@sutaakar Out of interest, why the need for this change?

Reply:
@robpblake The ContextClassLoader wasn't able to resolve the resource from uber-jar, I guess it doesn't have access to the main jar, probably just to its dependencies.
System ClassLoader can resolve resources from uber-jar.